EpicLoot/Data/EIDFLegacy.cs

```diff
@@ -11,30 +12,44 @@ public static class EIDFLegacy
         public const string StartDelimiterEscaped = "randyknapp.mods.extendeditemdataframework.startdelimiter";
         public const string EndDelimiterEscaped = "randyknapp.mods.extendeditemdataframework.enddelimiter";
 
+        public const string CrafterNameType = "c";
+        public const string UniqueIdType = "u";
+
+        public static bool ExtendedItemFrameworkLoaded = false;
+
         public static string RestoreDataText(string text)
         {
             return string.IsNullOrEmpty(text) ? text : text.Replace(StartDelimiterEscaped, StartDelimiter).Replace(EndDelimiterEscaped, EndDelimiter);
         }
 
         public static bool IsLegacyEIDFItem(this ItemDrop.ItemData itemData)
         {
+            if (itemData == null || string.IsNullOrEmpty(itemData.m_crafterName))
+                return false;
+
             return itemData.m_crafterName.StartsWith(StartDelimiter);
         }
         public static bool IsLegacyMagicItem(this ItemDrop.ItemData itemData)
         {
+            if (itemData == null || string.IsNullOrEmpty(itemData.m_crafterName))
+                return false;
+
             return itemData.m_crafterName.Contains($"{StartDelimiter}{MagicItemComponent.TypeID}{EndDelimiter}");
         }
 
-        public static string GetMagicItemFromCrafterName(ItemDrop.ItemData item)
+        private static string _getEIDFTypeValue(string encodedCrafterName, string typeID)
```

Comment:
strange function name, my style doesn't include decorating private function names, they should be the same as public